### PR TITLE
Fix JS tests on fresh `npm install`s

### LINKED ISSUE #2265: Failing JS unit tests when the node modules are installed from scratch
**Describe the bug**
A couple of JS unit tests that normally pass, fail when `package-lock.json` and `node-modules` are removed locally and added again.

**To Reproduce**
1. Pull the latest `develop`. It's at `4f47751f` as of now
2. Run `npm run test:js`
3. Notice that all tests pass. All good until here
4. Now delete the `package-lock.json` file
5. Delete the `node_modules` directory
6. Run `npm i`
7. Run `npm run test:js` again
8. Notice that now there are two tests failing: `PaymentRequestsButtonPreview › displays an info notice if stripe is falsy` and `PaymentRequestsButtonPreview › displays the payment button when stripe is loaded` from `client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js`

**Expected behavior**
I'd expect the tests to remain consistent before and after re-installing all of the packages since there weren't any changes.

**Environment (please complete the following information):**
 - Node version: 12.22.6
 - Npm version: 6.14.15

**Additional context**
There's some more context in the PR for setting up the e2e environment, [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2238). 

The issue seems to be that the speak function from @wordpress/a11y isn't being mocked [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2238/files#diff-f558a2bb1e6e3625f54d34e9b1225700a2c13a2b851220dad69fea4cf6298a70L7).

The speak function [prints the notice for screen readers](https://github.com/WordPress/gutenberg/blob/7cbba2646a522738fd95894db8240e4d3d5ac1d3/packages/a11y/src/index.js#L60), so we have the message twice on the page when Stripe isn't available (we use the [Notice](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/notice/index.js#L35) component to display the message). All good so far.

Because speak was mocked before, we're not expecting the screen reader version of this message in the tests, where they're being printed now, thus they fail.

It looked like this was happening because of the packages that were introduced there, but this also happens on `develop` so they aren't related. Hat tip to @diegocurbelo for taking the time to go through this.

### PR DESCRIPTION
**Pull request by asumaran**
Fixes #2265

The [first approach](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2280/commits/a1b2db79245f0799219b66fbc9b8345f44a95316) I did was to just mock the `Notice` component (which internally uses  `@wordpress/a11y`) in `@wordpress/components` but I wasn't too happy about it because the whole module was being replaced and not just the `Notice` component.

So then after a deeper look at it I noticed the error was happening because the `package-lock.json` re-ordered the dependency tree for `@wordpress/a11y`.

This is how the dependency tree looks for `@wordpress/a11y` in the `develop` branch.

```
~/www/woocommerce-gateway-stripe ‹develop› » npm ls @wordpress/a11y
woocommerce-gateway-stripe@6.0.0 /Users/asumaran/www/woocommerce-gateway-stripe
├─┬ @woocommerce/navigation@6.1.0
│ ├─┬ @wordpress/components@11.1.3
│ │ └── @wordpress/a11y@2.15.3
│ └─┬ @wordpress/notices@1.12.0
│   └── @wordpress/a11y@2.15.3
└─┬ @wordpress/components@15.0.0
  └── @wordpress/a11y@3.2.1
```

The `@wordpress/a11y@3.2.1` module used by `@wordpress/components@15.0.0` is located at the root of `node_modules`. The tests passed because the mocked module was at the root of the `node_modules` so mocking it with `jest.mock( '@wordpress/a11y', () => ... )` was correct as it replaced the expected module.

And this is how it looks the dependency tree for the `@wordpress/a11y` module when running `npm install` after deleting the `package-lock.json` file and the `node_module` folder

```
~/www/woocommerce-gateway-stripe ‹as-fix-js-tests› » npm ls @wordpress/a11y
woocommerce-gateway-stripe@6.0.0 /Users/asumaran/www/woocommerce-gateway-stripe
├─┬ @woocommerce/navigation@6.1.0
│ ├─┬ @wordpress/components@11.1.3
│ │ └── @wordpress/a11y@2.15.3
│ └─┬ @wordpress/notices@1.12.0
│   └── @wordpress/a11y@2.15.3  deduped
└─┬ @wordpress/components@15.0.0
  ├── @wordpress/a11y@3.2.4
  └─┬ @wordpress/rich-text@5.0.7
    └── @wordpress/a11y@3.2.4  deduped
```

The `@wordpress/a11y@3.2.4` module used by `@wordpress/components@15.0.0` is located at `./node_modules/@wordpress/components/node_modules/@wordpress/a11y`. The one located at the root of `node_modules` is `@wordpress/a11y@2.15.3` which is not used by `@wordpress/components@15.0.0`.

Adding the full path to the module we wanted to mock fixed the issue.

### CODE REVIEW

File: client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js [hunk 1]
```diff
@@ -4,8 +4,12 @@ import { useStripe } from '@stripe/react-stripe-js';
 import PaymentRequestsButtonPreview from '../payment-request-button-preview';
 import { shouldUseGooglePayBrand } from '../utils/utils';
 
-jest.mock( '@wordpress/a11y', () => ( {
-	...jest.requireActual( '@wordpress/a11y' ),
+// We need to mock the actual module being used by `<Notice />` in the `@wordpress/components` module
+const realPathToA11yModule =
+	'../../../../node_modules/@wordpress/components/node_modules/@wordpress/a11y';
```
Comment (ricardo) [inline client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js]: Nit: `@wordpress/components/node_modules/@wordpress/a11y` should work too. It's a bit shorter.

**Review by ricardo — APPROVED** [review]
LGTM.

I'm not very familiar with the context of this issue and the relationship between the e2e tests, so it would be nice to get a second look from @a-danae before it gets merged.

I don't know of any other ways to force Jest to use the right module. Perhaps an alternative approach to fix this could be removing the `speak` mocking and tweaking the test to be ok with multiple elements returning the expected text 🤔.

☝️ Just throwing some thoughts, not suggesting changes.

File: client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js (hunk 1 above)
Comment (asumaran) [reply client/entrypoints/payment-request-settings/__tests__/payment-request-button-preview.test.js]: done https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2280/commits/14bc2a4ae07d76a246507f9c03d568e7ca81eab4

**Comment by a-danae** [comment]
I was AFK last week, sorry and thank you both! 🙌

**Pull request merged**